cartridge/roles/coordinator.lua

```diff
@@ -284,13 +284,32 @@ end
 -- @treturn[2] table Error description
 local function appoint_leaders(leaders)
     checks('table')
+
+    local servers = vars.topology_cfg.servers
+    local replicasets = vars.topology_cfg.replicasets
+
     local updates = {}
     for k, v in pairs(leaders) do
         if type(k) ~= 'string' or type(v) ~= 'string' then
             error('bad argument #1 to appoint_leaders' ..
                 ' (keys and values must be strings)', 2
             )
         end
+
+        local replicaset = replicasets[k]
+        if replicaset == nil then
+            error('Replicaset ' .. k .. ' does not exist')
```

Comment:
It's not a contract violation, don't raise it. `return nil, AppintmentError:new()` instead

Reply:
Ok, fixed